src/publisher/api_v2_views.py

```diff
@@ -289,36 +302,40 @@ def article_fragment(request, msid, fragment_id):
         ensure(fragment_id not in reserved_keys, "that key is protected")
 
         if request.method == "POST":
-            fragment_logic.add_fragment_update_article(
-                article, fragment_id, request.data
-            )
+            if request.content_type != "application/json":
+                return error_response(
+                    415,
+                    "refused: unsupported data. content-type must be 'application/json'",
+                )
+            fragment = utils.json_loads(request.body)
```

Comment:
I guess it comes from here. Previously `request.data()` would convert it to json for us.

add test to capture what happens when bad json is posted